pkg/registry/generic/etcd/etcd.go

```diff
@@ -441,6 +450,9 @@ func (e *Etcd) Delete(ctx api.Context, name string, options *api.DeleteOptions)
 	// delete immediately, or no graceful deletion supported
 	out := e.NewFunc()
 	if err := e.Storage.Delete(ctx, key, out); err != nil {
+		if ignoreNotFound && storage.IsNotFound(err) {
+			return e.finalizeDelete(obj, true)
```

Comment:
> that sounds like another can of worm

Agree, I would not modify anything other than this particular error case at this point. Post-1.2, I think it would make sense to work on removing the entire update-followed-by-delete path for deletes with grace-period=0.

The last `existing` object that `GuaranteedUpdate` successfully persisted might be the most reasonable thing we could return here without reworking `GuaranteedUpdate`. Something like

```
var lastExisting runtime.Object = nil
var useLastExistingForNotFound bool = false
...
   lastGraceful = *options.GracePeriodSeconds
+ lastExisting = existing
   return existing, nil
...
case nil:
-  ignoreNotFound = true
+  useLastExistingForNotFound = true
...
if useLastExistingForNotFound && lastExisting != nil && storage.IsNotFound(err){
  return e.finalizeDelete(lastExisting, true)
}
```